packages/python-sqlalchemy/superfunctions_sqlalchemy/adapter.py

```
@@ -0,0 +1,584 @@
+                                model=normalized.model,
+                                data=row,
+                                select=normalized.select,
+                                namespace=normalized.namespace,
+                            ),
+                            conn,
+                        )
+                    )
+                return created_rows
+        
+        except IntegrityError as e:
+            raise ConstraintViolationError(str(e), cause=e)
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)